### PR TITLE
Static analysis

### PR DESCRIPTION
**Pull request by CarlaBuongiorno**

### CODE REVIEW

File: test_tortoise_interpreter.py [hunk 1]
```diff
@@ -4,45 +4,77 @@
...
+                "W 2  # draw west 2cm",
+                "N 1  # then north 1",
+                "E 2  # then east 2",
+                "S 1  # then back south",
+                "U    # pen up",
+            ],
+            [["P", "2"], ["D"], ["W", "2"], ["N", "1"], ["E", "2"], ["S", "1"], ["U"]],
+        ),
+    ],
+)
```
Comment (the-riviera-kid) [inline test_tortoise_interpreter.py]: If you think this formatting was better before, you can disable it with (I think) `# noqa`

File: tortoise.py [hunk 2]
```diff
@@ -1,23 +1,28 @@
+"""Mini-language to control a simple turtle-graphics system."""
+
 import sys
 import tortoise_interpreter
 
 
 def main():
+    """Read into a tortoise program to be interpreted."""
```
Comment (the-riviera-kid) [inline tortoise.py]: Typo: Should say "Read in a..."

File: tortoise_interpreter.py [hunk 3]
```diff
@@ -1,51 +1,77 @@
...
+        "W": move_west,
+    }
+    command_to_parse = navigate_command_functions(commands, program)
 
     for command in command_to_parse:
         print_to_screen(command)
 
 
 def sanitize_program(program):
+    """Get tortoise program ready for interpretation."""
```
Comment (the-riviera-kid) [inline tortoise_interpreter.py]: This docstring is a little vague; what makes it "ready for interpretation"? Maybe combine this one-line docstring with the larger comment below, and explain why this is required?

File: tortoise_interpreter.py [hunk 4]
```diff
@@ -1,51 +1,77 @@
...
             if len(command) > 2:
-                del command[command.index(' '):]
+                del command[command.index(" ") :]
     return program
 
 
-def navigate_command_functions(COMMANDS, program):
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+def navigate_command_functions(commands, program):
+    """Number each command of the tortoise program and call the
```
Comment (the-riviera-kid) [inline tortoise_interpreter.py]: Ideally, you want your first line of the docstring to be a short summary, and the details can go in the body. This is a bit long for one line.

File: tools.bash [hunk 5]
```diff
@@ -0,0 +1,10 @@
+echo
+echo ======================== PYTEST =======================
+pytest
+echo
+echo ======================== BLACK =======================
+black .
+echo
+echo ======================== PYLINT =======================
+pylint tortoise.py tortoise_interpreter.py
```
Comment (the-riviera-kid) [inline tools.bash]: We talked about this; should be globbed :)

**Review by the-riviera-kid — CHANGES_REQUESTED** [review]

File: test_tortoise_interpreter.py (hunk 1 above)
Comment (CarlaBuongiorno) [reply test_tortoise_interpreter.py]: `# fmt: off` and `# fmt: on` is the pylint version it seems.

File: tools.bash [hunk 6]
```diff
@@ -6,5 +6,5 @@ echo ======================== BLACK =======================
 black .
 echo
 echo ======================== PYLINT =======================
-pylint tortoise.py tortoise_interpreter.py
+pylint *.py
```
Comment (the-riviera-kid) [inline tools.bash:9]: :partying_face: :balloon:

**Review by the-riviera-kid — APPROVED** [review]

**Pull request merged**